src/store/index-level.ts

```diff
@@ -37,75 +33,124 @@ export class IndexLevel {
     return this.db.close();
   }
 
-  async put(id: string, entry: Entry, options?: IndexLevelOptions): Promise<void> {
-    entry = flatten(entry) as Entry;
-
-    const ops: LevelWrapperBatchOperation<string>[] = [ ];
-    const prefixes: string[] = [ ];
-    for (const property in entry) {
-      const value = entry[property];
-
-      const prefix = this.join(property, this.encodeValue(value));
-      ops.push({ type: 'put', key: this.join(prefix, id), value: id });
-      prefixes.push(prefix);
+  /**
+   * Adds indexes for a specific data/object/content.
+   * @param id ID of the data/object/content being indexed.
+   */
+  async put(
+    id: string,
+    indexes: { [property: string]: unknown },
+    options?: IndexLevelOptions
+  ): Promise<void> {
+
+    indexes = flatten(indexes);
+
+    const operations: LevelWrapperBatchOperation<string>[] = [ ];
+
+    // create an index entry for each property in the `indexes`
+    for (const propertyName in indexes) {
+      const value = indexes[propertyName];
```

Comment:
It took me a second to read properly that this `value` isn't the same as the `value: id` a couple line down. I'm trying to think of another variable name than `value` for this.

What do you think of renaming `value` to `propertyValue` or `indexValue`?

Reply:
Now that I think about it, I'm on the fence about these variable names too. I'll leave it up to you.

Reply:
`propertyValue` sounds good. Changing.